src/main/resources/templates/java/test/test/${packageNameFolder}/StructuralTest.java

```diff
@@ -6,76 +6,176 @@
+        if(observedModifiers.equals(new String[]{""}) && expectedModifiers.length() == 0) {
+            return true;
+        }
+
+        // If the number of the modifiers does not match, then the modifiers per se do not match either.
+        if(observedModifiers.length != expectedModifiers.length()) {
+            return false;
+        }
+
+        // Otherwise check if each expected modifier is contained in the array of the observed ones.
+        // If at least one isn't, then the modifiers don't match.
+        boolean allModifiersAreImplemented = true;
+        for(Object expectedModifier : expectedModifiers) {
+            allModifiersAreImplemented &= Arrays.asList(observedModifiers).contains((String) expectedModifier);
+        }
+
+        return allModifiersAreImplemented;
+    }
+
+    /**
+     * This method checks if the parameters of the actual structural element (in this case method or constructor)
+     * match the ones in the expected structural element.
+     * @param observedParameters: The actual parameter types as a classes array.
+     * @param expectedParameters: The expected parameter type names as a JSONArray.
+     * @return True if they match, false otherwise.
+     */
+    protected boolean checkParameters(Class<?>[] observedParameters, JSONArray expectedParameters) {
+
+        // If the number of parameters do not match, then the parameters per se do not match either.
+        if(observedParameters.length != expectedParameters.length()) {
+            return false;
+        }
+
+        // Create hash tables to store how often a parameter type occurs.
+        // Checking the occurrences of a certain parameter type is enough, since the parameter names
+        // or their order are not relevant to us.
+        String[] expectedParameterTypeNames = new String[expectedParameters.length()];
+        for(int i = 0; i < expectedParameters.length(); i++) expectedParameterTypeNames[i] = expectedParameters.getString(i);
+        HashMap<String, Integer> expectedParametersHashtable = createParametersHashMap(expectedParameterTypeNames);
+
+        String[] observedParameterTypeNames = new String[observedParameters.length];
+        for(int i = 0; i < observedParameters.length; i++) observedParameterTypeNames[i] = observedParameters[i].getSimpleName();
+        HashMap<String, Integer> observedParametersHashtable = createParametersHashMap(observedParameterTypeNames);
+
+        return expectedParametersHashtable.equals(observedParametersHashtable);
+    }
+
+    /**
+     * This method creates a hash table where the name of a parameter type is hashed to the number of the occurrences in the passed string collection.
+     * @param parameterTypeNames
+     * @return
+     */
+    private HashMap<String, Integer> createParametersHashMap(String... parameterTypeNames) {
+        HashMap<String, Integer> parametersHashTable = new HashMap<String, Integer>();
```

Comment:
On the left side you should always use the Interface Map instead of HashMap

```suggestion
        Map<String, Integer> parametersHashTable = new HashMap<String, Integer>();
```